### PR TITLE
Test: add ignored benchmark on a cluster of 3 nodes

### PR DESCRIPTION
**Pull request by drmingdrmer**
##### Test: add ignored benchmark on a cluster of 3 nodes

---

---
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/openraft/196)

### CODE REVIEW

File: openraft/tests/benchmark/bench_cluster.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        router.client_request(0, "foo", i as u64).await
+    }
+
+    let elapsed = now.elapsed();
+
+    println!(
+        "n={}, time: {:?}, op/s: {}",
+        n,
+        elapsed,
+        elapsed.as_nanos() / (n as u128)
```
Comment (schreter) [inline openraft/tests/benchmark/bench_cluster.rs:65]: This is computing ns/op, not ops/s.

```suggestion
        (n as u128 * 1_000_000_000) / elapsed.as_nanos()
```

**Comment by drmingdrmer** [comment]
@schreter

Does the setup in this benchmark reflect the one you mentioned in:
https://github.com/datafuselabs/openraft/pull/189#issuecomment-1047267421

I use only 1 task for `client_write`.
Do you have multiple threads for client write?

**Comment by schreter** [comment]
> I use only 1 task for client_write. Do you have multiple threads for client write?

I tested both with a single task pushing requests (this is primarily relevant for latency and seeing the scheduling issues in Tokio) and with multiple parallel tasks (relevant for throughput and seeing the throughput-limiting issues in openraft).

With all optimizations so far including my refactoring I did locally, I reached about 70K requests/second with a single task and about 450K requests/second with 16 tasks on my first-gen Apple M1 MBP (release build). On my "big" Linux machine, the throughput is only about 1/3rd of the MBP.

**Review by schreter — COMMENTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/openraft/196)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ariesdevil and @lichuang)

___
*[openraft/tests/benchmark/bench_cluster.rs, line 59 at r1](https://reviewable.io/reviews/datafuselabs/openraft/196#-Mw_SLnA0gjh6u2jbf11:-Mw_SLnA0gjh6u2jbf12:b-ixiyzw) ([raw file](https://github.com/datafuselabs/openraft/blob/34f5004c1af419fcaf0919102012e2144e03d951/openraft/tests/benchmark/bench_cluster.rs#L59)):*
> ```XML
> 
>     println!(
>         "n={}, time: {:?}, ns/op: {}",
> ```

Maybe you should also add the output of ops/sec. When testing for throughput with parallel tasks, ns/op will become "wrong", since there are multiple operations running in parallel.

**Review by xp-trumpet — APPROVED** [review]

**Review by ariesdevil — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/openraft/196)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @lichuang)

**Pull request merged**